### ISSUE TITLE
Incomplete ruv occasionally returned  from ruv search

### ISSUE THREAD

**Issue by jchapma** [389ds/389-ds-base#5193]
**Issue Description**
An intermittent condition occurs during cleanallruv (force) CI tests which results in an incomplete ruv being returned to the client. This generates an "IndexError" in lib389 because of the ruv->replica_purl being NULL. 

**Package Version and Platform:**
 - Platform: 5.6.6-300.fc32.x86_64 and 5.14.0-63.el9.x86_64
 - Package and version: 389-Directory/2.0.14 

**Steps to Reproduce**
1. Run cleanallruv CI test (This issue occurs occasionally, so you might need to run it more than once)
```
 sudo py.test -s -v dirsrvtests/tests/suites/replication/cleanallruv_test.py::test_clean_force
```

2. Notice CI test error
```
self = <lib389.replica.RUV object at 0x7f9ad7b5c2b0>
ruvs = ['{replicageneration} 6204409b000000010000', 
'{replica 1 ldap://julia32:39001} 6204409b000100010000 620440e90000000100...000 620440c5000000020000', '{replica 3 ldap://julia32:39003} 620440a6000100030000 620440c8000100030000', '{replica 4}']
logger = None

    def __init__(self, ruvs=[], logger=None):
        if logger is not None:
            self._log = logger
        else:
            self._log = logging.getLogger(__name__)
        self._rids = []
        self._rid_url = {}
        self._rid_rawruv = {}
        self._rid_csn = {}
        self._rid_maxcsn = {}
        self._rid_modts = {}
        self._data_generation = None
        self._data_generation_csn = None
        # Process the array of data
        for r in ruvs:
            pr = r.replace('{', '').replace('}', '').split(' ')
            if pr[0] == 'replicageneration':
                # replicageneration 5a2ffd0f000000010000
                self._data_generation = pr[1]
            elif pr[0] == 'replica':
                self._log.info("jc pr%s" % (pr))
                # replica 1 ldap://ldapkdc.example.com:39001 5a2ffd0f000100010000 5a2ffd0f000200010000
                # Don't add rids if they have no csn (no writes) yet.
                rid = pr[1]
                self._rids.append(rid)
>               self._rid_url[rid] = pr[2]
E               IndexError: list index out of range

/usr/local/lib/python3.8/site-packages/lib389-1.4.0.1-py3.8.egg/lib389/replica.py:853: IndexError
```
2.1 Notice an incomplete ruv for replica 4 while running an ruv search in parallel with CI test 
```
nsds50ruv: {replicageneration} 6225e1a3000000010000
nsds50ruv: {replica 1 ldap://julia32:39001} 6225e1a4000000010000 6225e20600000
 0010000
nsds50ruv: {replica 2 ldap://julia32:39002} 6225e1a6000100020000 6225e1cf00000
 0020000
nsds50ruv: {replica 3 ldap://julia32:39003} 6225e1af000100030000 6225e1d200000
 0030000
nsds50ruv: {replica 4}
```

**Expected results**
Ruv's returned from a ruv search should contain at least rid and replica_purl elements.

**Additional context**
The access logs confirm that there is an ext op add for the same entry twice, the first add works fine but the second add fails because of an existing entry uniqueid. During the second add, an ruv is created with an rid and replica_purl set to NULL, which is never updated and written to the changelog as the previous operation exits with LDAP_ALREADY_EXISTS.

**Comment by mreynolds389** [comment]
This needs to be cherry picked to master branch, 389-ds-base-2.1, 389-ds-base-1.4.4 and 389-ds-base-1.4.3.  Just curious why is this only against 389-ds-base-2.0?   Usually we start at master branch and cherry pick down to lower branches.  This "might" require a new PR for Master branch.

**Comment by jchapma** [comment]
> Just curious why is this only against 389-ds-base-2.0? Usually we start at master branch and cherry pick down to lower branches. This "might" require a new PR for Master branch.

I couldn't reproduce the IndexError exception on master for some reason ...

**Comment by progier389** [comment]
I think I fixed the Index Error at lib389 level a few months ago while fixing several failing CI test but anyway as Mark is saying we should align the code ...

**Comment by jchapma** [comment]
389-ds-base-1.4.3  cbadae9870a43ad120bdf9d7c26e1c28cb4b3c7b
389-ds-base-1.4.4  2741799a7929acd343c4571dbc6c926dbb99e0c0
389-ds-base-2.1     36338ab71ef5ac446db94d9dafa211f8e3a613a5
